src/Microsoft.Health.Fhir.Core/Features/Persistence/IFhirDataStore.cs

```diff
@@ -20,5 +20,11 @@ public interface IFhirDataStore
         Task<ResourceWrapper> GetAsync(ResourceKey key, CancellationToken cancellationToken);
 
         Task HardDeleteAsync(ResourceKey key, CancellationToken cancellationToken);
+
+        void BeginTransactionScope();
+
+        void CompleteTransactionScope();
+
+        void Dispose();
```

Comment:
I think if you have a `Dispose` method you should be taking a dependency on the IDisposable interface